### ISSUE TITLE
check if target os filesystem supports case-insensitive or not at build time is not correct

### ISSUE THREAD

**Issue by leleliu008** [universal-ctags/ctags#3026]
In our build script define a macro called `CASE_INSENSITIVE_FILENAMES`, this macro determined at build time. I think this logical is not correct. because many different filesystems can be used in one os. one filesystems is `case-insensitive`, another can be `case-sensitive`, so, this feature only can be determined at runtime.

**Comment by masatake** [comment]
You are correct. However, I am not positive about implementing it because all file systems (+ OS) I use are case-sensitive.
I want to take my time for improving the parsers and writing new parsers instead.

Unless the changes for this issue are not so complicated, I will not reject a pull request for this topic.

I wonder this one is related to this issue.
https://github.com/universal-ctags/ctags/issues/1216
